src/huggingface_hub/repository.py

```diff
@@ -344,6 +404,10 @@ def __init__(
         if revision is not None:
             self.git_checkout(revision, create_branch_ok=True)
 
+        # This ensures that all commands exit before exiting the Python runtime.
+        # This will ensure all pushes register on the hub, even if other errors happen in subsequent operations.
```

Comment:
What happens if user tries to kill the process (for example ctrl + d) while things are getting pushed?

Reply:
If the user tries to kill the process, it will try to continue pushing:
```py
>>> repo.command_queue
[[push command, status code: running, in progress. PID: 95581]]
>>> (CTRL+C)
Waiting for the following commands to finish before shutting down: [[push command, status code: running, in progress. PID: 95581]].
```
If you try to exit again:
```
^CError in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/home/lysandre/Workspaces/Python/huggingface_hub/src/huggingface_hub/repository.py", line 1141, in wait_for_commands
    time.sleep(1)
KeyboardInterrupt
```
and then you're out!

Reply:
That's perfect!